app/components/provider_interface/dsa_success_page_component.rb

```diff
@@ -0,0 +1,42 @@
+module ProviderInterface
+  class DsaSuccessPageComponent < ViewComponent::Base
+    attr_reader :provider_user, :provider_permission_setup_pending
+
+    def initialize(provider_user:, provider_permission_setup_pending:)
+      @provider_user = provider_user
+      @provider_permission_setup_pending = provider_permission_setup_pending
+    end
+
+    def success_message
+      if show_new_content?
+        t('.success_message')
+      else
+        t('.old.success_message')
+      end
+    end
+
+    def setup_permissions_message
+      if show_new_content?
+        t('.setup_permissions_message')
+      else
+        t('.old.setup_permissions_message')
+      end
+    end
+
+    def setup_permissions_action
+      if show_new_content?
+        t('.setup_permissions_action')
+      else
+        t('.old.setup_permissions_action')
+      end
+    end
+
+    def user_can_manage_users?
+      provider_user.authorisation.can_manage_users_for_at_least_one_provider?
+    end
+
+    def show_new_content?
+      @show_new_content ||= FeatureFlag.active?(:accredited_provider_setting_permissions)
+    end
```

Comment:
I thought about splitting the non-feature flag work into a separate component, but decided it was not worth it